### DISCUSSION TITLE
Multitask GP Regression tasks in a batch form？

### DISCUSSION THREAD

**Discussion by Rashfu** [cornellius-gp/gpytorch#2387] [Q&A]
I have read the tutorials on [Multitask GP Regression](https://docs.gpytorch.ai/en/stable/examples/03_Multitask_Exact_GPs/Multitask_GP_Regression.html) and [Batch GP Regression](https://docs.gpytorch.ai/en/stable/examples/08_Advanced_Usage/Simple_Batch_Mode_GP_Regression.html). I would like to know if it is possible to parallelize multiple Multitask GP Regression tasks in a batch form. For example, I have a dataset `(X, y)` with the shape :` torch.Size([4, 500, 3]), torch.Size([4, 500, 6])`. Here, 4 represents the batch size, 3 represents the input dimension, and 6 represents the output dimension.

**Comment by Balandat** [comment]
In general this should be possible. Taking a step back though, are the locations of the outputs in your case shared across all of the tasks? And if so, are your observations noisy or noiseless? If locations are shared and observations are noiseless, there is no benefit from using a multi-task model as it reduces to a set of independent models (due to something called autokrigability).

**Reply by Rashfu** [reply]
Thank you for your reply. I will provide a more detailed description of my question.


1. In each batch, the output `y` does not share the same positions `X`. In other words, the input positions `x` for the first `500×3` and the fourth `500×3` are different. So, it can be split into 4 independent regression tasks: `X (500,3)` → `y (500,6)`. However, I would like to know if it is possible to parallelize the computation in batch form.
2. In each independent task, y represents correlated output dimensions, meaning the 6 dimensions of the output have some similarities. Therefore, I have to use a multi-task model.
3. The last point I would like to clarify is that in my case, both `X`(locations) and `y`(observations) have unknown noise. However, in the tutorial, the part about uncertain inputs ([Exact GP](https://docs.gpytorch.ai/en/stable/examples/01_Exact_GPs/GP_Regression_DistributionalKernel.html#)) assumes that the variances are known. This confuses me a bit because I thought the variances should be parameters to be optimized in maximizing the marginal log likelihood.

I understand that the issue I presented is somewhat complex, and I appreciate your detailed response once again.
